pkg/crc/cluster/kubeadmin_password.go

```diff
@@ -11,47 +13,56 @@ import (
 	"github.com/code-ready/crc/pkg/crc/constants"
 	"github.com/code-ready/crc/pkg/crc/logging"
 	"github.com/code-ready/crc/pkg/crc/oc"
-	crcos "github.com/code-ready/crc/pkg/os"
 	"golang.org/x/crypto/bcrypt"
 )
 
-// UpdateKubeAdminUserPassword does following
-// - Create and put updated kubeadmin password to ~/.crc/machine/crc/kubeadmin-password
-// - Update the htpasswd secret
-func UpdateKubeAdminUserPassword(ocConfig oc.Config) error {
+// GenerateKubeAdminUserPassword creates and put updated kubeadmin password to ~/.crc/machine/crc/kubeadmin-password
+func GenerateKubeAdminUserPassword() error {
+	logging.Infof("Generating new password for the kubeadmin user")
 	kubeAdminPasswordFile := constants.GetKubeAdminPasswordPath()
-	if crcos.FileExists(kubeAdminPasswordFile) {
-		logging.Debugf("kubeadmin password has already been updated")
-		return nil
+	kubeAdminPassword, err := GenerateRandomPasswordHash(23)
+	if err != nil {
+		return fmt.Errorf("Cannot generate the kubeadmin user password: %w", err)
 	}
+	return ioutil.WriteFile(kubeAdminPasswordFile, []byte(kubeAdminPassword), 0600)
+}
 
-	logging.Infof("Generating new password for the kubeadmin user")
-
-	kubeAdminPassword, err := GenerateRandomPasswordHash(23)
+// UpdateKubeAdminUserPassword updates the htpasswd secret
+func UpdateKubeAdminUserPassword(ocConfig oc.Config) error {
+	kubeAdminPassword, err := GetKubeadminPassword()
 	if err != nil {
 		return fmt.Errorf("Cannot generate the kubeadmin user password: %w", err)
 	}
+	credentials := map[string]string{
+		"developer": "developer",
+		"kubeadmin": kubeAdminPassword,
+	}
 
-	hashDeveloperPasswd, err := hashBcrypt("developer")
+	given, _, err := ocConfig.RunOcCommandPrivate("get", "secret", "htpass-secret", "-n", "openshift-config", "-o", `jsonpath="{.data.htpasswd}"`)
 	if err != nil {
 		return err
 	}
-
-	hashKubeAdminPasswd, err := hashBcrypt(kubeAdminPassword)
+	ok, err := compareHtpasswd(given, credentials)
```

Comment:
This check is going to break the behavior of cluster if user do something following.
- Start crc 
- Add more user to htpasswd using https://github.com/code-ready/crc/wiki/Add-another-user-to-cluster
- Stop the cluster
- Start again and here we again check len of password on cluster and what we have here which will not match
- We force again to update the password which doesn't have user's specific password.

Reply:
Ok good point, I will try to write a merge function then.

Reply:
@guillaumerose Looking again to the code, why do you think we need that kind of complicated logic on crc side in `UpdateKubeAdminUserPassword` function? why is just checking the `kubeadmin_password` file in the `~/.crc/machine/crc/` directory is enough and return  in case it there. On openshift side it make sense because controller need to make sure the password updated in the resource should be updated but I think in crc this might not needed util you think current way have some breakage?

I do think generating this password file in early stage is a good improvement.

Reply:
The current mechanism differentiates how it works when creating the VM vs restarting the VM. 

```
if first run {
  change the password
} else {
  assume password is ok
} 
```

I think it is not a good thing to have lot of code only running for the first start: 
* Each crc start should ensure the cluster is a good shape. Right now, we don't know about the status of the auth at the 2nd run. Maybe the user broke it. With this change, we are sure the auth will work. 
* I did it already for the pull secret, cluster ID and I found it more robust against CTRL+C during the first start. Ideally I would like `machine.Start()` code to be idempotent (and remove the `if running { check nothing, assume it is ok, return }`.